### PR TITLE
feat: add Summary component

### PR DESCRIPTION
**Pull request by revolunet**

### CODE REVIEW

File: src/Summary.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+type SummaryLink = {
+    text: string;
+    linkProps: RegisteredLinkProps;
+};
+
+export type SummaryProps = {
+    className?: string;
+    links: SummaryLink[];
+    title?: string;
+    as?: "p" | "h2" | "h3" | "h4" | "h5" | "h6";
```
Comment (garronej) [inline src/Summary.tsx:21]:
```suggestion
    /** Default: "p" */
    as?: "p" | "h2" | "h3" | "h4" | "h5" | "h6";
```

File: src/Summary.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+/** @see <https://react-dsfr-components.etalab.studio/?path=/docs/components-summary> */
+export const Summary = memo(
+    forwardRef<HTMLDivElement, SummaryProps>((props, ref) => {
+        const { className, links, as = "p", title, classes = {}, ...rest } = props;
+
+        const { t } = useTranslation();
+
+        const titleId = useId();
+        const summaryTitle = title || t("title");
```
Comment (garronej) [inline src/Summary.tsx:33]:
```suggestion
        const summaryTitle = title ?? t("title");
```

File: src/Summary.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                        className: cx(fr.cx("fr-summary__title"), classes.title),
+                        id: titleId
+                    },
+                    <>{summaryTitle}</>
+                )}
+                <ol>
+                    {links.map(
+                        (link, idx) =>
+                            link.linkProps.href !== undefined && (
+                                <li key={link.linkProps.href + idx}>
```
Comment (garronej) [inline src/Summary.tsx:56]:
```suggestion
                                <li key={idx}>
```

File: src/Summary.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                                    <a
```
Comment (garronej) [inline src/Summary.tsx:57]: We should use the `Link` component:  

https://github.com/codegouvfr/react-dsfr/blob/1174b180494215be90f340f904f0027c0f52521b/src/Header/Header.tsx#L124

File: src/Summary.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                                        className={cx(fr.cx("fr-summary__link"), classes.link)}
+                                        {...link.linkProps}
```
Comment (garronej) [inline src/Summary.tsx:59]:
```suggestion
                                        {...link.linkProps}
                                        className={cx(fr.cx("fr-summary__link"), classes.link, link.linkProps.className)}
                                        
```

**Review by garronej — COMMENTED** [review]
Thank you <3

File: src/Summary.tsx (hunk 2 above)
Comment (revolunet) [reply src/Summary.tsx:34]: learning everyday :p thanks !

**Comment by revolunet** [comment]
PS: squash please

**Pull request merged**